drivers/mtd/mtd_partition.c

```diff
@@ -579,7 +579,7 @@ static ssize_t part_procfs_read(FAR struct file *filep, FAR char *buffer,
                   MTDIOC_GEOMETRY, (unsigned long)((uintptr_t)&geo));
           if (ret < 0)
             {
-              ferr("ERROR: mtd->ioctl failed: %d\n", ret);
+              ferr("ERROR: mtd->ioctl failed: %" PRId32 "\n", ret);
```

Comment:
```suggestion
              ferr("ERROR: mtd->ioctl failed: %zd\n", ret);
```
`ssize_t` (as well as `size_t`) are implementation-dependent, so we should not use a fixed-length specifier for them.
According to the C standard, the correct specifier for size types is **z**.